### PR TITLE
feat: PostHog setup and instrumentation

### PR DESCRIPTION
**Pull request by KristianSelnas**

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -78,13 +79,15 @@ export const App = () => {
...
                                   <AppLanguageProvider>
                                     <GeolocationContextProvider>
-                                      <GlobalMessagesContextProvider>
-                                        <BottomSheetProvider>
-                                          <FeedbackQuestionsProvider>
-                                            <RootStack />
-                                          </FeedbackQuestionsProvider>
-                                        </BottomSheetProvider>
-                                      </GlobalMessagesContextProvider>
+                                      <AnalyticsContextProvider>
```
Comment (gorandalum) [inline src/index.tsx:82]: Is it intentional that the `AnalyticsContextProvider` is added both here and under `NavigationContainer`?

File: src/analytics/AnalyticsContext.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      PostHog.initAsync(POSTHOG_API_KEY, {
+        host: POSTHOG_HOST,
+      }).then(setClient);
+    }
+  }, []);
+
+  // Since PostHog offers auto capture of navigation events,
+  // the PostHog provider expects to be setup inside a React Navigation object.
+  // In cases where the no navigation is available, e.g. the bottom sheet,
+  // the PostHog provider will throw an error.
```
Comment (rosvik) [inline src/analytics/AnalyticsContext.tsx:23]: Does this mean we won't be able to have any analytics in the bottom sheet?

**Review by rosvik — APPROVED** [review]
LGTM 👍

File: src/index.tsx (hunk 1 above)
Comment (KristianSelnas) [reply src/index.tsx:82]: Yes, It's necessary to get auto capture to work. Auto capture must be set up with navigation available.

**Pull request merged**